src/kdc/kdc_preauth.c

```diff
@@ -799,7 +790,7 @@ hint_list_finish(struct hint_state *state, krb5_error_code code)
     kdc_realm_t *kdc_active_realm = state->realm;
 
     if (!code) {
-        if (state->pa_data[0] == 0) {
+        if (state->pa_data == NULL) {
```

Comment:
This is checking a different thing.  That's intended?

Reply:
Yes; with new helper functions an empty pa_data list is represented as a null pointer, while the old code would have preallocated an array and just not filled any entries yet.